app/flatpak-builtins-utils.c

```diff
@@ -581,3 +581,159 @@ get_permission_tables (XdpDbusPermissionStore *store)
 
   return (char **) g_ptr_array_free (tables, FALSE);
 }
+
+/*** column handling ***/
+
+static int
+find_column (Column *columns,
+             const char *name,
+             GError **error)
+{
+  int i;
+  int candidate;
+
+  candidate = -1;
+  for (i = 0; columns[i].name; i++)
+    {
+      if (g_str_equal (columns[i].name, name))
+        {
+          return i;
+        }
+      else if (g_str_has_prefix (columns[i].name, name))
+        {
+          if (candidate == -1)
+            {
+              candidate = i;
+            }
+          else
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED, _("Ambiguous column: %s"), name);
+              return -1;
+            }
+        }
+    }
+
+  if (candidate >= 0)
+    return candidate;
+
+  g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED, _("Unknown column: %s"), name);
+  return -1;
+}
+
+static Column *
+column_filter (Column *columns,
+               const char *col_arg,
+               GError **error)
+{
+  g_auto(GStrv) cols = g_strsplit (col_arg, ",", 0);
+  int n_cols = g_strv_length (cols);
+  Column *sorted = g_new0 (Column, n_cols + 1);
+  int i;
+
+  for (i = 0; i < n_cols; i++)
+    {
+      int idx = find_column (columns, cols[i], error);
+      if (idx == -1)
+        return FALSE;
+      sorted[i] = columns[idx];
+    }
+
+  return sorted;
+}
+
+static gboolean
+list_has (const char *list,
+          const char *term)
+{
+  const char *p;
+  int len;
+
+  p = strstr (list, term);
```

Comment:
This needs to loop over strstr, or it will fail with list=foobar,bar and term=bar